standard/date_ext.lua

```diff
@@ -46,9 +49,10 @@ function DateExt.readTimestamp(dateString)
 
 	-- Extracts the '-4:00' out of <abbr data-tz="-4:00" title="Eastern Daylight Time (UTC-4)">EDT</abbr>
 	local tzTemplateOffset = dateString:match('data%-tz%=[\"\']([%d%-%+%:]+)[\"\']')
+	local tzAbbrName = dateString:match('[\"\']>(%w+)</abbr>')
 	local datePart = (mw.text.split(dateString, '<', true)[1]):gsub('-', '')
 	local timestampString = mw.getContentLanguage():formatDate('U', datePart .. (tzTemplateOffset or ''))
-	return tonumber(timestampString)
+	return tonumber(timestampString), tzTemplateOffset, tzAbbrName
```

Comment:
I think this should be split into `DateExt.readTimestamp` and `DateExt.readDateTime`. `readTimestamp` returns just the timestamp, and `readDateTime` returns `{timestamp = ..., tzOffset = ..., tzAbbr = ...}`, where `tzOffset` is `tzTemplateOffset` converted to seconds. (There should be better names than `readDateTime` - I can't come up with any atm)

The primary reason is to encourage callers to handle timezones properly in in code paths where it's important to keep the timezone. That means code paths which discard the timezone should use a different `read` function, lest people copy and paste code from timezone-free modules to timezone sensitive modules.

Reply:
I don't know how much time you want to spend on this - but it might be worth setting up a structural type for timestamp + timezone. It would have just two fields `{timestamp = ..., tzOffset = ...}`, and could be called `DateTime`. It would be passed around in lua as a whole, not two fields individually. It would only get split up when stored in lpdb, or when discarding the timezone. Functions which act on `DateTime` would go in `Module:DateTime`. 

`tzAbbr` is not included in `DateTime`, since it can be reverse-looked up from `Module:Timezone/Data`.

Reply:
tzAbbr can't be reverse looked up from offset as far as I can see. 
CET and BST (daylight for GMT) have the same offset +01:00. It would be confusing if we started writing CET for British Summer time, or use BST for Central European Winter time. Similar things apply with eg. China, Western Aus and Philipines, they all have the same timezone(+0800), but have different abbreviations

Reply:
Date object sounds sensible

Reply:
Let's make that a follow-up PR though.

Reply:
> Let's make that a follow-up PR though.

After look at this a bit yestarday, I agree. DateTime object would be great, but would significantly increase the size of the PR.